upgrade/steps.go

```diff
@@ -845,8 +845,8 @@ func migrationSteps(ctx Context, repo ProviderRepo, providerName string, descrip
 	return steps, nil
 }
 
-func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
-	providerName = strings.TrimPrefix(providerName, "pulumi-")
+func AddAutoAliasing(ctx Context, repo ProviderRepo) (step.Step, error) {
+	providerName := strings.TrimPrefix(repo.name, "pulumi-")
```

Comment:
It looks like we are still assuming that the provider's name will start with `pulumi-`, even though we let the user pass in information to the contrary: `iwahbe/my-pulumi-provider` as the repo name.

Reply:
Rebased on main now